iopool/pool.go

```diff
@@ -232,6 +156,12 @@ func (p *Pool) Get(ctx context.Context) (Entity, error) {
 		return e, err
 	}
 
+	t0 := time.Now()
+	checkout := func() {
+		p.metrics.getWait.Inc()
+		p.metrics.getWaitDuration.Add(time.Since(t0).Milliseconds())
+	}
```

Comment:
How about doing putting the `for` in a method and do something like 

```go
t0 := time.Now()
e, err := p.get(ctx)

if err == nil {
 p.metrics.getWait.Inc()
 p.metrics.getWaitDuration.Add(time.Since(t0).Milliseconds())
}

return e, err
```

Reply:
there multiple outcome and complicated branching. it will be hard to make it happen